pilot/pkg/serviceregistry/synthetic/serviceentry/syntheticserviceentrycontroller.go

```diff
@@ -122,6 +122,13 @@ func (c *SyntheticServiceEntryController) Apply(change *sink.Change) error {
 		return fmt.Errorf("apply: type not supported %s", change.Collection)
 	}
 
+	// Note: run asynchronously to not block mcp receive as it will not return any error
+	go c.handleChange(change)
```

Comment:
How are successive updates guaranteed to be ordered if the handling is aysnc? Its unlikely but still possible for a later update to be overwritten by a previous update.

Reply:
Yeah, good catch. I will introduce a queue here.

Reply:
The queue is unbounded. Alternatively, a buffered channel of size 1 would decouple stream.Recv and c.handleChange() while preserving backpreasure.